packit_service/worker/events/new_hotness.py

```diff
@@ -148,3 +147,58 @@ def get_dict(self, default_dict: Optional[Dict] = None) -> dict:
         result = super().get_dict(d)
         result.pop("_repo_url")
         return result
+
+
+# the decorator is needed in case the DB project event is not created (not valid arguments)
+# but we still want to report from pre_check of the PullFromUpstreamHandler
+@use_for_job_config_trigger(trigger_type=JobConfigTriggerType.release)
+class NewHotnessUpdateEvent(AnityaUpdateEvent):
+    def __init__(
+        self,
+        package_name: str,
+        version: str,
+        distgit_project_url: str,
+        bug_id: int,
+        release_monitoring_project_id: int,
+    ):
+        super().__init__(
+            package_name=package_name,
+            distgit_project_url=distgit_project_url,
+            release_monitoring_project_id=release_monitoring_project_id,
+        )
+        self._version = version
+        self.bug_id = bug_id
+
+    @property
+    def version(self) -> str:
+        return self._version
+
+
+@use_for_job_config_trigger(trigger_type=JobConfigTriggerType.release)
+class AnityaVersionUpdateEvent(AnityaUpdateEvent):
+    def __init__(
+        self,
+        package_name: str,
+        versions: list[str],
+        distgit_project_url: str,
+        release_monitoring_project_id: int,
+    ):
+        super().__init__(
+            package_name=package_name,
+            distgit_project_url=distgit_project_url,
+            release_monitoring_project_id=release_monitoring_project_id,
+        )
+
+        self._versions = versions
+
+    @property
+    def version(self) -> str:
```

Comment:
The major issue here is that we “generate” DB entry for the release from the event object. There's also serialization involved (for Celery).

At the same time it feels a bit weird to keep logic of “deciding the next version” in the »event«.

Reply:
discussed together with Mato that we want all the new versions, see e.g. https://apps.fedoraproject.org/datagrepper/v2/id?id=efec10ed-9f86-4e03-9e4f-2e626934b3ae&is_raw=true&size=extra-large in here we would be interested in all 3

Reply:
and then it is up to maintainer to define the per-branch jobs and rules (e.g. via version_udpate_mask, upstream_tag_include, upstream_tag_exclude): [example](https://src.fedoraproject.org/rpms/golang/blob/rawhide/f/.packit.yaml)